### ISSUE TITLE
fromError returns "Unknown error" for Javascript errors

### ISSUE THREAD

**Issue by guillaumebrunerie** [causaly/zod-validation-error#376]
Calling `fromError` on some standard Javascript error (for instance a `ReferenceError`) returns the message `Unknown error`. I think it should instead realize that the error has a `message` and use that.

**Comment by jmike** [comment]
This doesn't seem right.

We are handling standard JS errors correctly as you can see in https://github.com/causaly/zod-validation-error/blob/main/lib/toValidationError.ts#L15-L17

Furthermore, I added a PR to add tests for `ReferenceError` and `TypeError` in order to prove this is working correctly.
https://github.com/causaly/zod-validation-error/pull/377

Could you share a bit more info on how to replicate this issue?

**Comment by guillaumebrunerie** [comment]
Indeed, my bad. I think my issue was actually that I had a `ReferenceError` from a different realm (different `window` object), so the prototype was different and the `instanceof` check did not work. Unfortunately there is no standard `Error.isError` yet, but maybe `String(err)` could be used instead of `"Unknown error"`?

**Comment by jmike** [comment]
> Unfortunately there is no standard Error.isError yet, but maybe String(err) could be used instead of "Unknown error"?

Hm, let's investigate this together. These are the checks we are making (I am just copy-pasting from https://github.com/causaly/zod-validation-error/blob/main/lib/toValidationError.ts)...

```
// check if err is a ZodError
if (isZodErrorLike(err)) {
  return fromZodErrorWithoutRuntimeCheck(err, options);
}

// check if err is an instance of the native JS Error
if (err instanceof Error) {
  return new ValidationError(err.message, { cause: err });
}

// at this point Error can be anything
return new ValidationError('Unknown error');
```

Given that error can be anything (e.g. `number`, `Object`, `Map`) it's neither safe nor useful to convert it to `String`.

**Comment by jmike** [comment]
Closing this issue due to inactivity.
